### PR TITLE
First round of unittests for Altinn.App.Api

### PR DESCRIPTION
**Pull request by tjololo**
## Description
We are missing unittests for Altinn.App.Api. Adding the first round of tests, more will follow.

Restructuring the repository structure.
`/src/...` contains the source projects
`/test/...` contains the test projects

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: .github/workflows/test-and-analyze.yml [hunk 1]
```diff
@@ -49,8 +49,8 @@ jobs:
           dotnet tool update dotnet-sonarscanner --tool-path .\.sonar\scanner
       - name: Build & Test
         run: |
-           dotnet build src/AppRef.sln -v m
-           dotnet test src/AppRef.sln -v m
+           dotnet build AppLibDotnet.sln -v m
```
Comment (RonnyB71) [inline .github/workflows/test-and-analyze.yml:52]: Do we need this? According to the [docs](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-test) the test command will build before running tests.

File: test/Altinn.App.Api.Tests/Controllers/StatelessDataControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+
+namespace Altinn.App.Api.Tests.Controllers;
+
+public class StatelessDataControllerTests
+{
+    [Fact]
+    public async void Get_Returns_BadRequest_when_dataType_is_null()
+    {
+        // Arrange
+        var altinnAppMock = new Mock<IAltinnApp>();
```
Comment (RonnyB71) [inline test/Altinn.App.Api.Tests/Controllers/StatelessDataControllerTests.cs:32]: Any reason why you don't use WebApplicationFactory, request a client and use that to send the requests to the controller? That way you would include the route matching and you could create a custom implementation of your mock setup to get rid of all the mocks in each test.

**Review by RonnyB71 — APPROVED** [review]

File: .github/workflows/test-and-analyze.yml (hunk 1 above)
Comment (tjololo) [reply .github/workflows/test-and-analyze.yml:52]: Nice catch! I will update the workflow

File: test/Altinn.App.Api.Tests/Controllers/StatelessDataControllerTests.cs (hunk 2 above)
Comment (tjololo) [reply test/Altinn.App.Api.Tests/Controllers/StatelessDataControllerTests.cs:32]: Agreed, I need the new setup to be able to move forward with the move of the projects from app-template-dotnet repository.
We will need to add more test so I will make a note your suggestion and improve the tests in the next round of unittests

**Pull request merged**